source/plugins/ruby/in_kube_events.rb

```diff
@@ -153,7 +153,7 @@ def parse_and_emit_records(events, eventQueryState, newEventQueryState, batchTim
             next
           end
 
-          nodeName = items["source"].key?("host") ? items["source"]["host"] : (OMS::Common.get_hostname)
+          nodeName = items["source"].key?("host") ? items["source"]["host"] : ""
```

Comment:
This can break if the items["source"] doesnt exist. lets add null check around items["source"] for both places.